cartography/data/jobs/analysis/aws_foreign_accounts.json

```diff
@@ -0,0 +1,15 @@
+{
+  "statements": [
+    {
+      "__comment": "This analyze AWS accounts we created and tag the ones that are foreign. Foreign accounts are ones that were not in the sync scope",
+      "query": "MATCH (foreign:AWSAccount) where NOT EXISTS(foreign.inscope) SET foreign.foreign = true",
+      "iterative": false
+    },
+    {
+      "__comment": "Remove accounts that were set with foreign and inscope. This can happen as we finish the list of sync accounts through assume role mapping and vpc peering",
```

Comment:
Unlikely scenario, but what about when an account is now `foreign`, but used to be `inscope`? This query would incorrectly set the account as `inscope`.

We should maybe also be intentional to delete the `foreign` property if it exists.

Reply:
Case 1) Removal of foreign is done in aws_account_cleanup.json

Case 2) For the cases "an account is now foreign, but used to be inscope? " There might be other cases but I can see this happening if the user change profile and runs a scan on top of existing graph + the old AWS accounts are trusted by the new set of AWS account in the new profile.

We can wipe the inscope + foreign across all accounts at the beginning of the AWS module - That would impact anything that runs analysis using both foreign and inscope attributes during the full sync.

@ramonpetgrave64 Any other suggestions? Is it a blocker?

Reply:
I think my suggestion below in organizations.py covers the case. Let me know if that works.